cli/tests/activate.bats

```diff
@@ -3254,6 +3254,79 @@ EOF
 
 # ---------------------------------------------------------------------------- #
 
+# bats test_tags=activate,activate:attach
+# NB: There is a corresponding test in `services.bats`.
+@test "version: refuses to attach to an older activations.json version" {
+  project_setup
+
+  # TODO: Workaround for https://github.com/flox/flox/issues/2164
+  rm "${HOME}/.bashrc"
+
+  # Prevent backtraces from `flox-activations` leaking into output.
+  unset RUST_BACKTRACE
+
+  export -f jq_edit
+  FLOX_SHELL="bash" run "$FLOX_BIN" activate -- bash <(
+    cat << 'EOF'
+      echo "$PPID" > activation_pid
+
+      ACTIVATIONS_DIR=$(dirname "$_FLOX_ACTIVATION_STATE_DIR")
+      ACTIVATIONS_JSON="${ACTIVATIONS_DIR}/activations.json"
+      jq_edit "$ACTIVATIONS_JSON" '.version = 0'
+
+      "$FLOX_BIN" activate -- echo "should fail"
+EOF
+  )
+
+  # Capture from the previous activation.
+  ACTIVATION_PID=$(cat activation_pid)
+
+  assert_failure
+  refute_line "should fail"
+  assert_output "Error: This environment has already been activated with an older incompatible version of 'flox'
```

Comment:
nonblocking question: why isn't this getting the `❌ ERROR: failed to run activation script:` part of the error?

Reply:
`flox activate -- command` (Rust) execs into `activate` (Rust->Bash) so errors from `flox-activations` (Rust->Bash->Rust) don't get wrapped in `messages::error`. Whereas `flox services [re]start]` doesn't exec into `activate`. I originally put the emoji in `flox-activations` but it meant that we repeated the `❌ ERROR:` in one of the cases 😞